packages/cli/create-strapi-starter/utils/build-starter.js

```diff
@@ -12,28 +12,29 @@ const generateNewApp = require('@strapi/generate-new');
 
 const hasYarn = require('./has-yarn');
 const { runInstall, runApp, initGit } = require('./child-process');
-const { getRepoInfo, downloadGitHubRepo } = require('./fetch-github');
+const { getStarterPackageInfo, downloadNpmStarter } = require('./fetch-npm-starter');
 const logger = require('./logger');
 const stopProcess = require('./stop-process');
 
 /**
  * @param  {string} - filePath Path to starter.json file
  */
-function readStarterJson(filePath, starterUrl) {
+function readStarterJson(filePath, starter) {
   try {
     const data = fse.readFileSync(filePath);
     return JSON.parse(data);
   } catch (err) {
-    stopProcess(`Could not find ${chalk.yellow('starter.json')} in ${chalk.yellow(starterUrl)}`);
+    stopProcess(`Could not find ${chalk.yellow('starter.json')} in ${chalk.yellow(starter)}`);
   }
 }
 
 /**
  * @param  {string} rootPath - Path to the project directory
  * @param  {string} projectName - Name of the project
+ * @param  {boolean} useYarn - Use yarn instead of npm
  */
-async function initPackageJson(rootPath, projectName) {
-  const packageManager = hasYarn() ? 'yarn --cwd' : 'npm run --prefix';
+async function initPackageJson(rootPath, projectName, useYarn) {
```

Comment:
you should pass useYarn in an options object to make this function a bit more flexible